r2-netty/src/main/java/com/linkedin/r2/netty/client/http2/Http2ChannelLifecycle.java

```diff
@@ -186,11 +191,20 @@ private void notifyWaiters(Throwable e)
    */
   private void doBootstrapParentChannel(Callback<Channel> callback)
   {
-    _parentChannelLifecycle.create(new Callback<Channel>() {
+    // Lets not trust the _parentChannelLifecycle to timely return a response here.
+    // Embedding the callback inside a timeout callback (ObjectCreationTimeoutCallback)
+    // to force a response within creationTimeout deadline
+    _parentChannelLifecycle.create(new TimeoutCallback<>(_scheduler, _creationTimeout, TimeUnit.MILLISECONDS, new Callback<Channel>() {
       @Override
       public void onError(Throwable error)
       {
         callback.onError(error);
+
+        // Make sure to log the object creation timeout error
+        if (error instanceof ObjectCreationTimeoutException)
```

Comment:
Log error before callback.onError?

Reply:
This is intentionally placed after the callback to reduce any latency impact on the retriable error response as this request gets retried.

Reply:
ok